### ISSUE TITLE
[Nightly] Company Generator Creates Mechwarriors without mechs

### ISSUE THREAD

**Issue by repligator** [MegaMek/mekhq#4245]
### Environment
Build Date: 2024-06-16T23:38:20.419110358
Java Version: 17.0.11
Platform: Linux 6.5.0-35-generic (amd64)

### Description
When using the company generator, I received two pilots without mechs. I suspect that the following log entry had something to do with it. Will investigate when I get the chance.

`15:24:49,072 ERROR [mekhq.campaign.universe.generators.companyGenerators.AbstractCompanyGenerator] {AWT-EventQueue-0}
mekhq.campaign.universe.generators.companyGenerators.AbstractCompanyGenerator.generateEntity(AbstractCompanyGenerator.java:896) - Failed to generate an entity due to a null 'Mech summary for faction SL`

### Files
[mekhq.log](https://github.com/user-attachments/files/15905246/mekhq.log)
[mechless warriors zipped.gz.cpnx.zip](https://github.com/user-attachments/files/15905248/mechless.warriors.zipped.gz.cpnx.zip)

**Comment by Thom293** [comment]
I think its based on AtB rules, there is a chance a MW can roll as "dispossessed" without a mech.  Happens rarely, but it does happen.

**Comment by Thom293** [comment]
Here is a screen from the AtB doc and  from a very old spreadsheet that we had to use before MHQ was up to speed.  Someone automated the company generation.  

![image](https://github.com/MegaMek/mekhq/assets/23645569/3367d587-1463-47cc-a9bc-c7109745cefe)


![image](https://github.com/MegaMek/mekhq/assets/23645569/ee712d5b-543a-41a6-b785-755b06a9719e)

**Comment by repligator** [comment]
Looks like it occurs if the faction rolls as Star League, but Star League rolls false. It's also basically the same error you get if you ask for IS Omni-Mech's in 3025.

![Screenshot_20240619_161309](https://github.com/MegaMek/mekhq/assets/114337591/7658ae91-2c7b-4d2a-b08d-680774d75b04)

**Comment by IllianiCBT** [comment]
Yeah, this happens a lot if you play in very early eras, as a lot of factions don't have mechs in every category. I do wonder if we need to implement a fallback faction - IS for Inner Sphere, Clan to Clan, Periphery for Periphery, etc. Assuming that's not already implemented and it's just not triggering.

**Comment by IllianiBird** [comment]
As of 51.0 the current company generator is now officially deprecated and no further work will be done to fix or improve it. Please see the below announcement for more information :)

<img width="979" height="381" alt="Image" src="https://github.com/user-attachments/assets/ca39029d-6fa0-4d89-96cb-3689a4ee944c" />
